todo_app/app.py

```diff
@@ -117,20 +127,24 @@ def _index():
     @login_required
     def _add():
         if is_permitted('writer'):
+            logger.info("Adding task: %s " % request.form)
             todoapi.add_item(request.form)
+            
         return render_index_response()
 
     @app.route('/setstatus/<id>/<status>', methods=['POST'])
     @login_required
     def _setstatus(id,status):
         if is_permitted('writer'):
+            logger.info("Changing task %s to %s" % (id, status))
```

Comment:
You are successfully logging in a JSON format but not really taking advantage of structured logging. It would be nicer if this generated a log like `{"message": "Task status changed", "task_id": "abc", "new_status": "Done"}`

You can do this with the pythonjsonlogger by calling e.g. `logger.info('your message', extras={"task_id": id})`. Though you'll need to update your Logger wrapper class to pass it through.

Having structured logs like this would let you query your logs more easily
e.g. you could find every time any card was updated to Done without complicated/slow parsing of strings.